libcloud/common/base.py

```diff
@@ -867,6 +867,8 @@ def request(self, action, params=None, data=None, headers=None,
         return response
 
     def morph_action_hook(self, action):
+        if not action.startswith("/"):

```

Comment:
I still have an issue with this IF the host provided by the user surely that would be a request with a double slash?

http://example.com/ - host
containers/ - action

http://example.com//containers - resulting URL?

Can you try this and see what happens, if so this logic should check the last character in the host string as well